<h1 style="text-align:center; color:#005bbd; font-size:30px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
   Basic Types (Primitive Data Types)
</h1>

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul>
            <li>Basic types wo primitive data types hain jo C language ke foundation hain. Inke bina kisi bhi complex data type ko define karna mushkil hota hai. </li><br>
            <li>Ye types directly hardware ke sath compatible hoti hain aur memory mein specific size occupy karti hain. Basic types mein commonly used types char, int, float, aur bool hain.</li>
</ul></div></body>



<img style="display: block;margin-left: auto;margin-right: auto;width: 80%;" src="./images/primitive.jpg">

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>Integer</h3>
            <li>Integer data type whole numbers ko store karte hain.</li><br>
            <li>C programming mein integer data types ko memory ke size aur range ke hisaab se define kiya jata hai, aur yeh sign aur unsigned form mein hote hain</li>
</ul></div></body>



<h1 style="text-align:center; color:#005bbd; font-size:30px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
   Signed Integers
</h1>

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>Signed Integers (+/- values) :</h3>
            <li>Inka range negative aur positive dono numbers tak hota hai.</li>
            <li>Example: -10, 0, 25</li>
            <li>Data types: int, short int, long int, long long int</li>
</ul></div></body>



<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>short int</h3>
            <li>short int ko is tarah design kiya gaya hai ke yeh memory kam use kare aur sirf chhoti values ko store kare.</li><br>
            <li> Yeh usually 2 bytes (16 bits) ka memory space leta hai, lekin kuch systems par yeh 4 bytes bhi ho sakta hai (platform-specific differences hain). Iski range uske sign pe depend karti hai:</li><br>
            <li>Format Specifier: %hd</li>
</ul></div></body>



<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul>
Check karte hain ke mera system ki short int ki range kya hai
</ul></div></body>



In [24]:
#include <stdio.h>
#include <limits.h>

int main() {
    printf("Minimum value: %d\n", SHRT_MIN);
    printf("Maximum value: %d\n", SHRT_MAX);
    printf("Size of short int: %zu bytes", sizeof(short int));
    return 0;
}

Minimum value: -32768
Maximum value: 32767
Size of short int: 2 bytes

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul>
            <li>SHRT_MIN aur SHRT_MAX ko hum limits.h se access karte hain, jo short int ki range define karte hain.</li>
        <li>sizeof(short int) se short int ka size bytes mein milta hai.</li>
            <li>%zu format specifier sizeof ke output ko sahi tarah se display karta hai.</li>
</ul></div></body>



<img style="display: block;margin-left: auto;margin-right: auto;width: 60%;" src="./images/signint.jpg">

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul>
            Values store karne ke liye short int keyword ka use kiya jata hai.
</ul></div></body>



In [26]:
#include <stdio.h>
int main() {
    short int x = 32767;
    short int y = -32768;
    printf("x = %hd\ny = %hd",x,y);
    return 0;
}

x = 32767
y = -32768

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul>
Yahan hum zyada range ki value store karke dekhte hain ke kya value 16-bit se badi range ki value store hoti hai ya nahi.</ul></div></body>



In [16]:
#include <stdio.h>
int main() {
    short int x = -32769;
    printf("x = %hd",x);
    return 0;
}

warning: overflow in conversion from ‘int’ to ‘short int’ changes value from ‘-32769’ to ‘32767’ [-Woverflow],

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        value overflow ho gayi hai, iska matlab hai ke bit ki range 16 thi, lekin humne yahan 16-bit range se zyada data diya, jiski wajah se overflow yaani buffer overflow ho gaya.</div></body>



<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>int</h3>
            <li>int zyada commonly use hota hai, kyun ke iski range wide hoti hai aur yeh larger values handle kar sakta hai. </li><br>
            <li> int 4 bytes (32 bits) hota hai, yani ke short int se zyada range hoti hai.But Range mein (platform-specific differences hain)</li><br>
                        <li>Format Specifier: %d</li>

</ul></div></body>



In [29]:
#include <stdio.h>
#include <limits.h>

int main() {
    printf("Minimum value: %d\n", INT_MIN);
    printf("Maximum value: %d\n", INT_MAX);
    printf("Size of short int: %zu bytes", sizeof(int));
    return 0;
}

Minimum value: -2147483648
Maximum value: 2147483647
Size of short int: 4 bytes

In [7]:
#include <stdio.h>
int main() {
    int x = 2147483647;
    int y = -2147483648;
    printf("x = %d\ny = %d",x,y);
    return 0;
}

x = 2147483647
y = -2147483648

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>long int</h3>
            <li>long int data type ko design kiya gaya hai taake yeh aur bhi larger values ko store kar sake jo ke int se bhi zyada ho sakti hain.  </li><br>
            <li> Yeh 4 bytes ya 8 bytes ka memory space le sakta hai, jo platform-specific hota hai.</li><br>
                        <li>Format Specifier: %ld</li>

</ul></div></body>



In [51]:
#include <stdio.h>
#include <limits.h>

int main() {
    printf("Minimum value of signed long int: %ld\n", LONG_MIN);
    printf("Maximum value of signed long int: %ld\n", LONG_MAX);
    printf("Size of signed long int: %zu bytes\n", sizeof(long int));

    return 0;
}

Minimum value of signed long int: -9223372036854775808
Maximum value of signed long int: 9223372036854775807
Size of signed long int: 8 bytes


In [5]:
#include <stdio.h>

int main() {
    long int x = 9223372036854775807L;   // Max value for signed long int
    long int y = -9223372036854775807L - 1L; // Min value using subtraction
    
    printf("x =  %ld\n", x);
    printf("y = %ld\n", y);

    return 0;
}

x =  9223372036854775807
y = -9223372036854775808


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        C language me -9223372036854775808 ko directly likhne par kuch compilers error throw karte hain ya incorrect value interpret karte hain. <br>Yeh is liye hota hai kyunki C compiler pehle 9223372036854775808 ko positive value ki tarah parse karta hai, jo signed long int ke maximum range se zyada hai, aur uske baad - sign apply karta hai.<br><br>
Is problem ko solve karne ke liye hum -9223372036854775807L - 1L ka trick use karte hain, jo safe aur consistent tariqa hai minimum value define karne ka.<ul>
        <li>L ek suffix hai jo integer literals ko long ya long long type specify karne ke liye use hota hai.</li></ul>

</div></body>



<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>long long int</h3>
            <li>long long int data type ko C language mein aur bhi badi values ko store karne ke liye design kiya gaya hai, jo long int se bhi zyada ho sakti hain.</li><br>
            <li>Ye type 64-bit systems par mostly 8 bytes (64 bits) space leta hai aur ye large range of integers ko hold kar sakta hai. But actual value platform-specific hoti hai </li><br>
                        <li>Format Specifier: %lld</li>

</ul></div></body>



In [25]:
#include <stdio.h>
#include <limits.h>

int main() {
    printf("Minimum value of signed long long int: %lld\n",(long long) LLONG_MIN);
    printf("Maximum value of signed long long int:  %lld\n", LLONG_MAX);
    printf("Size of signed long long int: %zu bytes\n", sizeof(long long int));

    return 0;
}

Minimum value of signed long long int: -9223372036854775808
Maximum value of signed long long int:  9223372036854775807
Size of signed long long int: 8 bytes


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <p>C language mein data type sizes aur ranges platform dependent hain, yaani 32-bit aur 64-bit architectures pe data type ki sizes aur limits different ho sakti hain. <br>64-bit systems mein long int aur long long int dono 8 bytes ke hote hain, aur isi liye unki values same aati hain.</p><br>
Agar aap yehi code ek 32-bit system par run karenge, toh:<br>long int ka size 4 bytes hoga, aur maximum value 2147483647 hogi.<br>long long int ka size 8 bytes hoga, aur maximum value 9223372036854775807 hogi.<br><br>

Yani 32-bit systems pe long int aur long long int mein clear difference hoga.

</div></body>



In [28]:
#include <stdio.h>
#include <limits.h>

int main() {
    printf("Maximum value of signed long int:      %ld\n", LONG_MAX);
    printf("Maximum value of signed long long int: %lld\n", LLONG_MAX);

    return 0;
}


Maximum value of signed long int:      9223372036854775807
Maximum value of signed long long int: 9223372036854775807


In [29]:
#include <stdio.h>

int main() {
    long long int x = 9223372036854775807L;   // Max value for signed long int
    long long int y = -9223372036854775807L - 1L; // Min value using subtraction
    
    printf("x =   %lld\n", x);
    printf("y =  %lld\n", y);

    return 0;
}

x =   9223372036854775807
y =  -9223372036854775808


<h1 style="text-align:center; color:#005bbd; font-size:30px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
   UnSigned Integers
</h1>

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>Unsigned Integers (Only + values):</h3>
            <li>Inka range sirf positive numbers aur zero tak hota hai, negative values ko ye store nahi kar sakte.</li>
            <li>Example: 0, 25, 50</li>
            <li>Data types: unsigned int, unsigned short int, unsigned long int, unsigned long long int</li>
</ul></div></body>



<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>unsigned short int</h3>
            <li>unsigned short int ko is tarah design kiya gaya hai ke yeh memory kam use kare aur sirf chhoti values ko store kare.</li><br>
            <li> Yeh usually 2 bytes (16 bits) ka memory space leta hai, lekin kuch systems par yeh 4 bytes bhi ho sakta hai (platform-specific differences hain). Iski range uske sign pe depend karti hai:</li><br>
            <li>Format Specifier: %hd</li>
</ul></div></body>



In [34]:
#include <stdio.h>
#include <limits.h>

int main() {
    printf("Minimum value of unsigned short int: %u\n", 0); // Always 0 for unsigned types
    printf("Maximum value of unsigned short int: %u\n", USHRT_MAX);
    printf("Size of unsigned short int: %zu bytes\n", sizeof(unsigned short int));

    return 0;
}

Minimum value of unsigned short int: 0
Maximum value of unsigned short int: 65535
Size of unsigned short int: 2 bytes


In [40]:
#include <stdio.h>

int main() {
    unsigned short int x = 65535;
    unsigned short int y = 0;     
    
    printf("x = %hu\n", x); 
    printf("y = %hu\n", y);

    return 0;
}


x = 65535
y = 0


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>unsigned int</h3>
            <li>unsigned int data type negative values ko store nahi karta.  </li><br>
            <li> Yeh sirf non-negative integers ko store karne ke liye design kiya gaya hai, jiski wajah se iska maximum value signed int se zyada hota hai.</li><br>
            <li>Memory Size: unsigned int ka size bhi 4 bytes (32 bits) hota hai, jo ke platform-specific hai, lekin yeh hamesha non-negative values ko store karta hai.</li><br>
                        <li>Format Specifier: %u</li>

</ul></div></body>



In [41]:
#include <stdio.h>
#include <limits.h>

int main() {
    printf("Minimum value of unsigned int: %u\n", 0); 
    printf("Maximum value of unsigned int: %u\n", UINT_MAX);
    printf("Size of unsigned int: %zu bytes\n", sizeof(unsigned int));
    return 0;
}


Minimum value of unsigned int: 0
Maximum value of unsigned int: 4294967295
Size of unsigned int: 4 bytes


In [46]:
#include <stdio.h>

int main() {
    unsigned int x = 4294967295;
    unsigned int y = 0;     
    
    printf("x = %u\n", x); 
    printf("y = %u\n", y);

    return 0;
}

x = 4294967295
y = 0


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>unsigned long int</h3>
            <li>unsigned long int data type negative values ko store nahi karta. Yeh sirf non-negative integers ko store karne ke liye design kiya gaya hai, jiski wajah se iska maximum value signed long int se zyada hota hai.</li><br>
            <li> Memory Size: unsigned long int ka size platform-specific hota hai. Yeh 4 bytes ya 8 bytes le sakta hai. Typically, modern platforms par yeh 8 bytes hota hai.</li><br>
                        <li>Format Specifier: %lu</li>

</ul></div></body>



In [50]:
#include <stdio.h>
#include <limits.h>

int main() {
    printf("Minimum value of unsigned long int: %d\n", 0);
    printf("Maximum value of unsigned long int: %lu\n", ULONG_MAX); 
    printf("Size of unsigned long int: %zu bytes\n", sizeof(unsigned long int));
    return 0;
}


Minimum value of unsigned long int: 0
Maximum value of unsigned long int: 18446744073709551615
Size of unsigned long int: 8 bytes


In [55]:
#include <stdio.h>

int main() {
    unsigned long int x = 18446744073709551615UL;
    unsigned long int y = 0;     
    
    printf("x = %lu\n", x); 
    printf("y = %lu\n", y);

    return 0;
}


x = 18446744073709551615
y = 0


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
       UL ek suffix hai jo integer literals ko unsigned long ya unsigned long long type specify karne ke liye use hota hai.
</div></body>



<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 18px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        <ul> <h3>unsigned long int</h3>
            <li>unsigned long long int data type bhi negative values ko store nahi karta. Yeh sirf non-negative integers ko store karne ke liye design kiya gaya hai, jiski wajah se iska maximum value signed long long int se bhi zyada hota hai.</li><br>
            <li> Memory Size: unsigned long long int ka size hamesha 8 bytes hota hai, jo 64 bits ke liye sufficient hai.</li><br>
                        <li>Format Specifier: %llu</li>

</ul></div></body>



In [58]:
#include <stdio.h>
#include <limits.h>

int main() {
    printf("Minimum value of unsigned long long int: %d\n", 0); 
    printf("Maximum value of unsigned long long int: %llu\n", ULLONG_MAX); 
    printf("Size of unsigned long long int: %zu bytes\n", sizeof(unsigned long long int)); 
    return 0;
}

Minimum value of unsigned long long int: 0
Maximum value of unsigned long long int: 18446744073709551615
Size of unsigned long long int: 8 bytes


In [61]:
#include <stdio.h>

int main() {
    unsigned long long int x = 18446744073709551615UL;
    unsigned long long int y = 0;     
    
    printf("x = %llu\n", x); 
    printf("y = %llu\n", y);

    return 0;
}

x = 18446744073709551615
y = 0


<div style="color: black; font-size: 10x; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
    <h3 style="text-align:center;">Integers in C Ranges</h3>
</div>
<table style="width: 100%; border-collapse: collapse; margin: 20px 0; color: black; font-size: 10x; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px; table-layout: auto;">
    <thead>
        <tr style="background-color: #f2f2f2;">
            <th style="border: 1px solid black; padding: 8px; text-align: center;">Data Type</th>
            <th style="border: 1px solid black; padding: 8px; text-align: center;">Range</th>
            <th style="border: 1px solid black; padding: 8px; text-align: center;">Minimum Value</th>
            <th style="border: 1px solid black; padding: 8px; text-align: center;">Maximum Value</th>
            <th style="border: 1px solid black; padding: 8px; text-align: center;">Size (bytes)</th>
            <th style="border: 1px solid black; padding: 8px; text-align: center;">Format Specifier</th>
            <th style="border: 1px solid black; padding: 8px; text-align: center;">Suffix</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td style="border: 1px solid black; padding: 8px; text-align: center; max-width: 150px; word-wrap: break-word;">short int</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center; max-width: 150px; word-wrap: break-word;">-32,768 to 32,767</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">-32,768</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">32,767</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">2</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">%hd</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">N/A</td>
        </tr>
        <tr>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">unsigned short int</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">0 to 65,535</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">0</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">65,535</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">2</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">%hu</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">N/A</td>
        </tr>
        <tr>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">int</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">-2,147,483,648 to 2,147,483,647</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">-2,147,483,648</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">2,147,483,647</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">4</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">%d</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">N/A</td>
        </tr>
        <tr>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">unsigned int</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">0 to 4,294,967,295</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">0</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">4,294,967,295</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">4</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">%u</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">N/A</td>
        </tr>
        <tr>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">long int</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">-2,147,483,648 to 2,147,483,647 (32-bit),<br> -9,223,372,036,854,775,808 to 9,223,372,036,854,775,807 (64-bit)</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">-2,147,483,648</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">2,147,483,647 (32-bit)</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">4 or 8</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">%ld</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">N/A</td>
        </tr>
        <tr>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">unsigned long int</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">0 to 4,294,967,295 (32-bit),<br> 0 to 18,446,744,073,709,551,615 (64-bit)</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">0</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">4,294,967,295 (32-bit)</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">4 or 8</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">%lu</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">N/A</td>
        </tr>
        <tr>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">long long int</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">-9,223,372,036,854,775,808 to 9,223,372,036,854,775,807</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">-9,223,372,036,854,775,808</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">9,223,372,036,854,775,807</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">8</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">%lld</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">LL</td>
        </tr>
        <tr>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">unsigned long long int</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">0 to 18,446,744,073,709,551,615</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">0</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">18,446,744,073,709,551,615</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">8</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">%llu</td>
            <td style="border: 1px solid black; padding: 8px; text-align: center;">ULL</td>
        </tr>
    </tbody>
</table>


In [130]:
#include <stdio.h>
#include <limits.h>
int main(){
    printf("===============================================================\n");
    printf("\tsigned   short int Max            %d\n",SHRT_MAX);
    printf("\tsigned   short int Min           %d\n",SHRT_MIN);
    
    printf("\tunsigned short int Max            %d\n",USHRT_MAX);
    printf("\tunsigned short int Min            %d\n",0);
    
    printf("\tSize of signed short int          %zu\n",sizeof(short int));
    printf("\tSize of unsigned short int        %zu\n",sizeof(unsigned short int));
    printf("===============================================================\n");
    
    printf("\tsigned        int Max             %d\n",INT_MAX);
    printf("\tsigned        int Min            %d\n",INT_MIN);
    
    printf("\tunsigned      int Max             %u\n",UINT_MAX);
    printf("\tunsigned      int Min             %d\n",0);

    printf("\tSize of signed int                %zu\n",sizeof(int));
    printf("\tSize of unsigned int              %zu\n",sizeof(unsigned int));
    printf("===============================================================\n");
    
    printf("\tsigned   long int Max             %ld\n",LONG_MAX);
    printf("\tsigned   long int Min            %ld\n",LONG_MIN);
    
    printf("\tunsigned long int Max             %lu\n",ULONG_MAX);
    printf("\tunsigned long int Min             %d\n",0);

    printf("\tSize of signed long int           %zu\n",sizeof(long int));
    printf("\tSize of unsigned long int         %zu\n",sizeof(unsigned long int));
    printf("===============================================================\n");
    
    printf("\tsigned   long long int Max             %lld\n",LLONG_MAX);
    printf("\tsigned   long long int Min            %lld\n",LLONG_MIN);
    
    printf("\tunsigned long long int Max             %llu\n",ULLONG_MAX);
    printf("\tunsigned long long int Min             %d\n",0);

    printf("\tSize of signed long long int           %zu\n",sizeof(long long int));
    printf("\tSize of unsigned long long int         %zu\n",sizeof(unsigned long long int));
    printf("===============================================================");
    
        
    
}

	signed   short int Max            32767
	signed   short int Min           -32768
	unsigned short int Max            65535
	unsigned short int Min            0
	Size of signed short int          2
	Size of unsigned short int        2
	signed        int Max             2147483647
	signed        int Min            -2147483648
	unsigned      int Max             4294967295
	unsigned      int Min             0
	Size of signed int                4
	Size of unsigned int              4
	signed   long int Max             9223372036854775807
	signed   long int Min            -9223372036854775808
	unsigned long int Max             18446744073709551615
	unsigned long int Min             0
	Size of signed long int           8
	Size of unsigned long int         8
	signed   long long int Max             9223372036854775807
	signed   long long int Min            -9223372036854775808
	unsigned long long int Max             18446744073709551615
	unsigned long long int Min             0
	Size of signed l